python/dgl/transforms/functional.py

```diff
@@ -164,6 +166,14 @@ def knn_graph(x, k, algorithm='bruteforce-blas', dist='euclidean'):
         * 'euclidean': Use Euclidean distance (L2 norm) :math:`\sqrt{\sum_{i} (x_{i} - y_{i})^{2}}`.
         * 'cosine': Use cosine distance.
         (default: 'euclidean')
+    exclude_self : bool, optional
+        If True, the output graph will not contain self loop edges, and each node will not
+        be counted as one of its own k neighbors.  If False, the output graph will contain
+        self loop edges, and a node will be counted as one of its own k neighbors.
+    output_batch : bool, optional
+        If True, the output will be a batched graph representing just 1 KNN graph if :attr:`x`
+        is 2D, or representing :math:`shape(x)[0]` KNN graphs if :attr:`x` is 3D, as described
+        above.  If False, the output graph will be unbatched.
```

Comment:
"unbatch" means splitting a given graph into a list of smaller ones. How about "If False, the output graph will be a single graph with multiple components".

Reply:
Okay, fair enough.  "single graph with multiple components" also seems confusing, though, since in both cases, the return value is a single graph object with multiple components.  The only difference is whether the batch information is set.  That's why the first version of this description described the difference in terms of batch information being set or not.  I've committed a change that hopefully reconciles the concerns.  It's getting very verbose, though, for such a simple parameter that just controls whether batch information is set or not.